mqtt-streaming/src/test/scala/docs/scaladsl/MqttFlowSpec.scala

```diff
@@ -51,19 +52,23 @@ class MqttFlowSpec
       //#run-streaming-flow
       val (commands, events) =
         Source
-          .queue(3, OverflowStrategy.fail)
+          .queue(2, OverflowStrategy.fail)
           .via(mqttFlow)
-          .drop(3)
+          .collect {
+            case Right(Event(p: Publish, _)) => p
+          }
           .toMat(Sink.head)(Keep.both)
           .run()
 
       commands.offer(Command(Connect(clientId, ConnectFlags.CleanSession)))
       commands.offer(Command(Subscribe(topic)))
-      commands.offer(Command(Publish(topic, ByteString("ohi"))))
+      session ! Command(
+        Publish(ControlPacketFlags.RETAIN | ControlPacketFlags.QoSAtLeastOnceDelivery, topic, ByteString("ohi"))
+      )
       //#run-streaming-flow
 
       events.futureValue match {
-        case Right(Event(Publish(_, `topic`, _, bytes), _)) => bytes shouldBe ByteString("ohi")
+        case Publish(_, `topic`, _, bytes) => bytes shouldBe ByteString("ohi")
         case e => fail("Unexpected event: " + e)
       }
     }
```

Comment:
Other than the `session ! Command`, this is mostly about cleaning the test up and making it more resilient to out-of-expected-sequence `PubAck` messages.